Program.cs

```diff
@@ -53,10 +52,11 @@ class Program {
                if (j != 7) Console.WriteLine ("├───┼───┼───┼───┼───┼───┼───┼───┤");
             }
             Console.WriteLine ("└───┴───┴───┴───┴───┴───┴───┴───┘");
-            var k = Console.ReadKey (true);
-            while (k.Key != ConsoleKey.RightArrow && k.Key != ConsoleKey.LeftArrow) k = Console.ReadKey (true);
-            if (k.Key == ConsoleKey.LeftArrow && i > 0) i--;
-            else if (k.Key == ConsoleKey.RightArrow && i < sFinalSoln.Count) i++;
+            ConsoleKeyInfo k;
+            do {
+               k = Console.ReadKey (true);
+               if (k.Key == ConsoleKey.LeftArrow) { if (i > 0) i--; } else i++;

```

Comment:
Kindly test with different scenarios before submitting.

Here pressing other keys increment the variable 'i'.